labonneboite/web/api/views.py

```diff
@@ -266,15 +295,51 @@ def get_location(request_args):
     return location, zipcode, commune_id
 
 
-def create_fetcher(location, request_args):
+def create_visible_market_fetcher(request_args):
+    UNSUPPORTED_PARAMETERS = ['sort', 'naf_codes', 'headcount', 'departments', 'longitude', 'latitude']
+    MANDATORY_PARAMETERS = ['rome_codes', 'commune_id', 'contract']
+
+    for param in UNSUPPORTED_PARAMETERS:
+        if request_args.get(param):
+            raise InvalidFetcherArgument('parameter %s is not supported' % param)
+
+    for param in MANDATORY_PARAMETERS:
+        if not request_args.get(param):
+            raise InvalidFetcherArgument('parameter %s is required' % param)
+
+    commune_id = request_args.get('commune_id')
+
+    romes = [code.upper() for code in request_args.get('rome_codes').split(',')]
+    validate_rome_codes(romes)
+
+    page, page_size = get_page_and_page_size(request_args)
+    if page != 1:
+        raise InvalidFetcherArgument('only page=1 is supported as pagination is not implemented')
+
+    distance = get_distance(request_args)
+
+    contract = request_args.get('contract')
+    if contract == hiring_type_util.CONTRACT_ALTERNANCE:
+        hiring_type = hiring_type_util.CONTRACT_TO_HIRING_TYPE[contract]
+    else:
+        raise InvalidFetcherArgument('only contract=alternance is supported')
```

Comment:
Je pense que la formulation suivante est plus future-proof : "Invalid 'contract' parameter value. Only supported value is 'alternance'"

Reply:
euh mouais bof ça me parait cohérent avec tous les autres `InvalidFetcherArgument` - je vois pas le pb en fait

Reply:
ok